lib/RouteStore.js

```diff
@@ -0,0 +1,108 @@
+var createStore = require('fluxible/utils/createStore');
+var Router = require('routr');
+var queryString = require('query-string');
+var searchPattern = /\?([^\#]*)/;
+var util = require('util');
+var objectAssign = require('object-assign');
+var Immutable = require('immutable');
+
+var RouteStore = createStore({
+    storeName: 'RouteStore',
+    handlers: {
+        'NAVIGATE_START': 'handleNavigateStart',
```

Comment:
we should think about how RouteStore reacts to `NAVIGATE_SUCCESS` and `NAVIGATE_FAILURE` events.